src/lib/operators/jit_operator/operators/jit_aggregate.hpp

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include "abstract_jittable_sink.hpp"
+
+namespace opossum {
+
+// Represents an aggregate expression computed by the operator.
+// The tuple_value provides the values that are to be aggregated.
+// The resulting aggregates for each group of tuples are stored in the hashmap_value.
+// The optional hashmap_value_2 is used for computing averages. In this case the aggregate operator computes two
+// aggregates (a SUM and a COUNT) and divides the two in a post-processing step.
+// Each aggregate and groupby column stores its position in the output table. This allows the operator to maintain
+// arbitrary orders of groupby and aggregate columns.
+struct JitAggregateColumn {
+  std::string column_name;
+  uint64_t table_position;
+  AggregateFunction function;
+  JitTupleValue tuple_value;
+  JitHashmapValue hashmap_value;
+  std::optional<JitHashmapValue> hashmap_value_2 = std::nullopt;
+};
+
+// Represents a column that is considered when grouping tuples.
+// The tuple_value provides the values to group by.
+// The value for each group are stored in the hashmap_value.
+// Each aggregate and groupby column stores its position in the output table. This allows the operator to maintain
+// arbitrary orders of groupby and aggregate columns.
+struct JitGroupByColumn {
+  std::string column_name;
+  uint64_t table_position;
+  JitTupleValue tuple_value;
+  JitHashmapValue hashmap_value;
+};
+
+/* The JitAggregate operator performs materialization and must thus be the last operator in an operator chain.
+ * The operator groups tuples by a number of columns and computes a set of aggregates of each tuple group.
```

Comment:
You are not defining the term "tuple group", and more surprisingly, I did not find it anywhere in the code.

Reply:
I thought the term was pretty standard when describing aggregate operations. I changed the wording slightly. Were you only confused by the term or is the overall explanation unclear?

Reply:
It is more the fact that such a fundamental concept doesn't seem to appear anywhere by name in the code. If there was something named "tuple group" in the code, it would make it easier to link comment and code.